src/lib/sedlexing.ml

```diff
@@ -299,22 +311,16 @@ module Utf8 = struct
     (* http://www.faqs.org/rfcs/rfc3629.html *)
 
     let width =
-      let width = Array.make 256 (-1) in
-      let () =
-        for i = 0 to 127 do
-          width.(i) <- 1
-        done;
-        for i = 192 to 223 do
-          width.(i) <- 2
-        done;
-        for i = 224 to 239 do
-          width.(i) <- 3
-        done;
-        for i = 240 to 247 do
-          width.(i) <- 4
-        done
+      let a =
+        Array.init 256 (fun x ->
+            match Char.chr x with
+              | '\000' .. '\127' -> 1
+              | '\192' .. '\223' -> 2
+              | '\224' .. '\239' -> 3
+              | '\240' .. '\247' -> 4
+              | _ -> -1)
```

Comment:
🔥

Reply:
Actually now that I'm thinking about it, why not simply a function? 😅

```ocaml
let width = function
  | '\000' .. '\127' -> 1
  | '\192' .. '\223' -> 2
  | '\224' .. '\239' -> 3
  | '\240' .. '\247' -> 4
  | _ -> assert false

Reply:
I wanted to preserve the existing ideal to not affect perf but I don't think it should matter much. I've pushed the change.
We need to keep the `-1` for malformed input.